contracts/sol6/KyberStorage.sol

```diff
@@ -66,10 +67,28 @@ contract KyberStorage is IKyberStorage, PermissionGroupsNoModifiers, Utils5 {
         onlyAdmin();
         require(_kyberNetwork != IKyberNetwork(0), "network 0");
         emit KyberNetworkUpdated(_kyberNetwork);
-        previousNetworks.push(kyberNetwork);
+        networkHistory.saveContract(address(_kyberNetwork));
         kyberNetwork = _kyberNetwork;
     }
 
+    function setHistoricalContracts(
+        IKyberHistory _networkHistory,
+        IKyberHistory _feeHandlerHistory,
+        IKyberHistory _kyberDAOHistory,
+        IKyberHistory _matchingEngineHistory
+    ) external {
+        onlyAdmin();
+        require(_networkHistory != IKyberHistory(0), "networkHistory 0");
+        require(_feeHandlerHistory != IKyberHistory(0), "feeHandlerHistory 0");
+        require(_kyberDAOHistory != IKyberHistory(0), "kyberDAOHistory 0");
+        require(_matchingEngineHistory != IKyberHistory(0), "matchingEngineHistory 0");
+
+        networkHistory = _networkHistory;
```

Comment:
seems like this should be a one time call
I mean.
what motivation do we have to replace if it already stores some values.

if we do replace it.
we loose some trust for this list - from reserves / users.